streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryWindowStore.java

```diff
@@ -131,6 +149,11 @@ public void put(final Bytes key, final byte[] value, final long windowStartTimes
                 });
             }
         }
+
+        if (stateStoreContext != null && stateStoreContext.recordMetadata().isPresent()) {
```

Comment:
Optimistically updating position. Position makes sure we never move back. Or should we move this into the "value != null" case?

Reply:
I think we'll still want to advance the position, even if we otherwise ignore the record.